src/main/java/com/hivemq/cli/commands/options/TlsOptions.java

```diff
@@ -138,85 +137,298 @@ private boolean useBuiltSslConfig() {
-                    defaultCLIProperties.getServerCertificateChain();
-            if (defaultServerCertificate != null) {
-                if (serverCertificateChain == null) {
-                    serverCertificateChain = new ArrayList<>();
+    private @NotNull KeyManagerFactory createKeyManagerFactoryFromKeystore(
+            final @NotNull Path clientKeystore,
+            final @Nullable String clientKeystorePassword,
+            final @Nullable String clientKeystorePrivateKeyPassword)
+            throws NoSuchAlgorithmException, KeyStoreException, IOException, CertificateException,
+            UnrecoverableKeyException {
+        final KeyManagerFactory keyManagerFactory =
+                KeyManagerFactory.getInstance(KeyManagerFactory.getDefaultAlgorithm());
+        if (TlsUtil.isPKCS12Keystore(clientKeystore)) {
+            final KeyStore keyStore = KeyStore.getInstance("PKCS12");
+            final char[] keystorePassword;
+            if (clientKeystorePassword != null) {
+                keystorePassword = clientKeystorePassword.toCharArray();
+            } else {
+                keystorePassword = PasswordUtils.readPassword("Enter keystore password:");
+            }
+            try (final InputStream inputStream = Files.newInputStream(clientKeystore, StandardOpenOption.READ)) {
+                keyStore.load(inputStream, keystorePassword);
+            }
+            keyManagerFactory.init(keyStore, keystorePassword);
+        } else if (TlsUtil.isJKSKeystore(clientKeystore)) {
+            final KeyStore keyStore = KeyStore.getInstance("JKS");
+            final char[] keystorePassword;
+            if (clientKeystorePassword != null) {
+                keystorePassword = clientKeystorePassword.toCharArray();
+            } else {
+                keystorePassword = PasswordUtils.readPassword("Enter keystore password:");
+            }
+            try (final InputStream inputStream = Files.newInputStream(clientKeystore, StandardOpenOption.READ)) {
+                keyStore.load(inputStream, keystorePassword);
+            }
+
+            boolean isDifferentPrivateKeyPassword = false;
+            try {
+                keyManagerFactory.init(keyStore, keystorePassword);
+            } catch (final UnrecoverableKeyException differentPrivateKeyPassword) {
+                isDifferentPrivateKeyPassword = true;
+            }
+            if (isDifferentPrivateKeyPassword) {
+                final char[] keystorePrivateKeyPassword;
+                if (clientKeystorePrivateKeyPassword != null) {
+                    keystorePrivateKeyPassword = clientKeystorePrivateKeyPassword.toCharArray();
+                } else {
+                    keystorePrivateKeyPassword = PasswordUtils.readPassword("Enter keystore private key password:");
                 }
-                serverCertificateChain.addAll(defaultServerCertificate);
+                keyManagerFactory.init(keyStore, keystorePrivateKeyPassword);
```

Comment:
This is weird but I guess this is required because you give the option to type in that password.